### PR TITLE
Rename kotlin errors (#442)

### PR DESCRIPTION
**Pull request by bendk**

### CODE REVIEW

File: examples/arithmetic/tests/bindings/test_arithmetic.kts [hunk 1]
```diff
@@ -6,7 +6,7 @@ assert(add(4u, 8u) == 12uL)
 try {
     sub(0u, 2u)
     throw RuntimeException("Should have thrown a IntegerOverflow exception!")
-} catch (e: ArithmeticErrorException) {
+} catch (e: ArithmeticException) {
```
Comment (rfk) [inline examples/arithmetic/tests/bindings/test_arithmetic.kts:9]: OK yes, I'm one line in to the PR and already this seems much nicer 👍🏻

File: uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs [hunk 2]
```diff
@@ -280,4 +280,20 @@ mod filters {
             _ => format!("{}.read({})", type_kt(type_)?, nm),
         })
     }
+
+    /// Generate a Kotlin exception name
+    ///
+    /// This replaces "Error" at the end of the name with "Exception".  Rust code typically uses
+    /// "Error" for any type of error but in the Java word, "Error" means a non-recoverable error
```
Comment (rfk) [inline uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs:287]: sp: "word" => "world"

File: uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs [hunk 3]
```diff
@@ -280,4 +280,20 @@ mod filters {
...
+    /// and is distinguished from an "Exception".
+    pub fn exception_name_kt(error: &Error) -> Result<String, askama::Error> {
+        match error.name().strip_suffix("Error") {
+            None => Ok(error.name().to_owned()),
+            Some(trimmed) => {
+                let mut rv = trimmed.to_owned();
+                rv.push_str("Exception");
+                Ok(rv)
```
Comment (rfk) [inline uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs:295]: Purely as a style-consistency nit, I'm not 100% sure how to parse "rv" here - is it "return value"? I think I'd lightly prefer a more specific name like `err_name` or similar, but I don't feel strongly about it (and I'm sure there are plenty of things in this codebase that are not consistent with variable names 😅)

**Review by rfk — APPROVED** [review]
Great, thanks for diving in! I left a couple of style nits, feel free to take 'em or leave 'em.

Please also add a pending entry at the top of the changelog, calling out in particular that this is a breaking change for consumers.

File: uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs (hunk 3 above)
Comment (bendk) [reply uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs]: Yeah, "rv" was supposed to mean return value.  I just pushed a new commit that has something more descriptive.

**Pull request merged**